lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -591,6 +610,25 @@ class ReportUIFeatures {
     const popup = window.open(url, windowName);
   }
 
+  /**
+   * Opens a new tab to an external page and sends data via base64 encoded url params.
+   * @param {{lhr: LH.Result} | LH.Treemap.Options} data
+   * @param {string} url_
+   * @param {string} windowName
+   * @protected
+   */
+  static openTabWithUrlData(data, url_, windowName) {
+    const url = new URL(url_);
+    const params = new URLSearchParams();
+    for (const [key, value] of Object.entries(data)) {
+      params.set(key, btoa(JSON.stringify(value)));
+    }
+    url.hash = params.toString();
```

Comment:
maybe I'm missing the reason for it, but this seem like a complicated way to do this? Would stringify (the whole options object) -> base64 encode -> escape `+`, `/`, and `=` then set it as the hash be easier? It would also save the decode having to mess around with `URLSearchParams` for the hash, too

Reply:
`btoa` is going to have issues with UTF-8, too. I don't know if there's a modern way to deal with that (`TextEncoder`?) but the old way was `encodeURIComponent`/`decodeURIComponent`

Reply:
this guy did all the hard research: https://stackoverflow.com/a/30106551/2788187

just a dozen lines.

![image](https://user-images.githubusercontent.com/4071474/118742281-f5e90080-b804-11eb-91de-fa66274e9a6a.png)

it works.